src/sentry/issues/grouptype.py

```diff
@@ -43,6 +70,62 @@ def __post_init__(self) -> None:
         if self.category not in valid_categories:
             raise ValueError(f"Category must be one of {valid_categories} from GroupCategory.")
 
+        # ensure that noise config ignore limits only increase with rollouts
+        # use a ratio of ignore limit to expiry time to allow for greater flexibility
+        if self.group_policy:
+            group_policy = self.group_policy
+
+            if group_policy.feature:
+                try:
+                    features.get(group_policy.feature, None)
+                except FeatureNotRegistered:
+                    raise ValueError(f"Feature {group_policy.feature} is not registered")
+
+            prev_ignore_limit_ratio = 0.0
+            if group_policy.limited_access:
+                prev_ignore_limit_ratio = (
+                    group_policy.limited_access.ignore_limit
+                    / group_policy.limited_access.expiry_time
+                )
+
+            if group_policy.early_access:
+                early_access_ratio = (
+                    group_policy.early_access.ignore_limit / group_policy.early_access.expiry_time
+                )
+                if early_access_ratio < prev_ignore_limit_ratio:
+                    raise ValueError(
+                        "Early Access ignore limit ratio must be greater than Limited Access ignore limit ratio"
+                    )
+                prev_ignore_limit_ratio = early_access_ratio
+
+            general_access_ratio = (
+                group_policy.general_access.ignore_limit / group_policy.general_access.expiry_time
+            )
+            if general_access_ratio < prev_ignore_limit_ratio:
+                raise ValueError(
+                    "General Access ignore limit ratio must be greater than Early Access and Limited Access ignore limit ratios"
+                )
+
+
+def get_noise_config(
+    grouptype: Type[GroupType], organization: Organization
+) -> Optional[NoiseConfig]:
+    group_policy = grouptype.group_policy
+    if not group_policy:
+        return None
+
+    if (
+        group_policy.feature
+        and features.has(group_policy.feature, organization, actor=None)
+        and group_policy.limited_access
+    ):
+        return group_policy.limited_access
+
+    if organization.flags.early_adopter and group_policy.early_access:
+        return group_policy.early_access
+
+    return group_policy.general_access
```

Comment:
If we rolled this out to all orgs, wouldn't all orgs be in GA at that point? Or I guess what I'm wondering is how we'd move an org that was in LA to use the EA noise config, and eventually the GA one?

Reply:
The idea is that you'd just tinker with the noise config values, though I guess LA orgs are not necessarily contained in the EA org set